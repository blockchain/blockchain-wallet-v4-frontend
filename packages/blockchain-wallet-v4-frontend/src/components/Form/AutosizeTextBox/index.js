import React from 'react'
import { AutosizeTextInput, Text } from 'blockchain-info-components'
import styled from 'styled-components'

const Container = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;
  height: 40px;
`
const Error = styled(Text)`
  position: absolute;
  display: block;
  height: 15px;
  top: ${props => (props.errorBottom ? '40px' : '-20px')};
  right: 0;
`
const getErrorState = ({ invalid, touched }) => {
  return touched && invalid ? 'invalid' : 'initial'
}

class AutosizeTextBox extends React.Component {
  render () {
    const {
      autoComplete,
      autoFocus,
      borderRightNone,
      center,
      className,
      disabled,
      errorBottom,
      hideError,
      input,
      maxLength,
      meta,
      noLastPass,
      placeholder
    } = this.props
    const { active, error, initial, touched, warning } = meta
    const errorState = getErrorState(meta)

    return (
      <Container className={className}>
        <AutosizeTextInput
          {...input}
          autoFocus={autoFocus}
          autoComplete={autoComplete}
          borderRightNone={borderRightNone}
          active={active}
          disabled={disabled}
          errorState={errorState}
          initial={initial}
          placeholder={placeholder}
          center={center}
          noLastPass={noLastPass}
          maxLength={maxLength}
          data-e2e={this.props['data-e2e']}
        />
        {touched && error && !hideError && (
          <Error
            size='12px'
            weight={500}
            color='error'
            errorBottom={errorBottom}
          >
            {error}
          </Error>
        )}
        {touched && !error && warning && (
          <Error
            size='12px'
            weight={500}
            color='error'
            errorBottom={errorBottom}
          >
            {warning}
          </Error>
        )}
      </Container>
    )
  }
}

export default AutosizeTextBox
