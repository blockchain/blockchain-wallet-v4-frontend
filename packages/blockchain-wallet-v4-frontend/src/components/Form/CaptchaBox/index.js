import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { actions } from 'data'
import { getData } from './selectors'
import Error from './template.error'
import Loading from './template.loading'
import Success from './template.success'

class CaptchaBoxContainer extends React.PureComponent {
  constructor (props) {
    super(props)
    this.fetchNewCaptcha = this.fetchNewCaptcha.bind(this)
  }

  componentDidMount () {
    this.props.actions.fetchCaptcha()
  }

  fetchNewCaptcha () {
    this.props.actions.fetchCaptcha()
  }

  render () {
    const { data } = this.props

    return data.cata({
      Success: value => (
        <Success
          captchaUrl={value.url}
          fetchNewCaptcha={this.fetchNewCaptcha}
          {...this.props}
        />
      ),
      Failure: message => <Error>{message}</Error>,
      NotAsked: () => <Loading />,
      Loading: () => <Loading />
    })
  }
}

const mapStateToProps = state => ({
  data: getData(state)
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions.core.data.misc, dispatch)
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CaptchaBoxContainer)
