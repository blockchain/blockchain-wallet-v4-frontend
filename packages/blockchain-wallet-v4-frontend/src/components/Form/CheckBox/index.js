import React from 'react'
import styled from 'styled-components'

import { CheckBoxInput, Text } from 'blockchain-info-components'

const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
`
const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  height: 25px;
`
const Error = styled(Text)`
  display: block;
  height: 15px;
`
const getErrorState = meta => {
  return meta.touched && meta.invalid ? 'invalid' : 'initial'
}

const CheckBox = ({ children, className, ...field }) => {
  const errorState = getErrorState(field.meta)
  const checked = field.input.value || false

  return (
    <Wrapper className={className}>
      <Container className='Container'>
        <CheckBoxInput
          {...field.input}
          disabled={field.disabled}
          checked={checked}
          errorState={errorState}
        >
          {children}
        </CheckBoxInput>
      </Container>
      {field.meta.touched &&
        field.meta.error &&
        !field.hideErrors && (
          <Error size='12px' weight={300} color='error'>
            {field.meta.error}
          </Error>
        )}
    </Wrapper>
  )
}

export default CheckBox
