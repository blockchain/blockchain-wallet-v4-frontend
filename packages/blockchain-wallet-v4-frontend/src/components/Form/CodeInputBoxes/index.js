import React from 'react'
import styled from 'styled-components'
// import { Text } from 'blockchain-info-components'

const Container = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;
  width: 100%;
  height: 40px;
`
// const Error = styled(Text)`
//   position: absolute;
//   display: block;
//   top: 40px;
//   left: 0;
//   height: 15px;
// `
// const getErrorState = (meta) => {
//   return !meta.touched ? 'initial' : (meta.invalid ? 'invalid' : 'valid')
// }

const CodeInputBoxes = field => {
  // const errorState = getErrorState(field.meta)

  return (
    <Container>
      <span>code input boxes</span>
      {/* <TextInput {...field.input} errorState={errorState} placeholder={field.placeholder} /> */}
      {/* {field.meta.touched && field.meta.error && <Error size='12px' weight={300} color='error'>{field.meta.error}</Error>} */}
    </Container>
  )
}

export default CodeInputBoxes
