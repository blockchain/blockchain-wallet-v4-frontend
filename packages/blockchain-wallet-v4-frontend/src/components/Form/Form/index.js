import React from 'react'
import { Form as ReduxForm } from 'redux-form'
import styled from 'styled-components'

const BaseForm = styled(ReduxForm)`
  width: 100%;
`

const Form = props => {
  const { children, className, onSubmit } = props

  return (
    <BaseForm className={className} onSubmit={onSubmit}>
      {children}
    </BaseForm>
  )
}

export default Form
