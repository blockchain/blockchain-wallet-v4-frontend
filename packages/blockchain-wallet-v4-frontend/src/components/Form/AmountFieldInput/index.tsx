import React, { FocusEventHandler, useCallback, useRef, useState } from 'react'
import { Row } from 'blockchain-wallet-v4-frontend/src/modals/Swap/EnterAmount/Checkout'
import { Field } from 'redux-form'
import styled from 'styled-components'

import Currencies from '@core/exchange/currencies'
import { Icon, Text } from 'blockchain-info-components'
import { AmountTextBox } from 'components/Exchange'
import { useCoinConfig } from 'hooks'
import { formatTextAmount } from 'services/forms'

const AmountContainer = styled.div`
  display: flex;
  flex-direction: column;
  position: relative;
  width: 100%;
`

const AmountRow = styled(Row)<{ isError: boolean }>`
  position: relative;
  padding: 12px;
  justify-content: center;
  align-items: center;
  border: 0;
  width: 100%;

  & #amount-row {
    max-width: 90%;
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;

    & > input {
      color: ${(props) => (props.isError ? 'red400' : 'textBlack')};
    }
  }
`

const QuoteRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`

const AmountFieldInput: React.FC<Props> = ({
  amountError,
  coin,
  fiatCurrency,
  fix,
  name,
  onChange,
  onToggleFix,
  quote,
  showCounter,
  showToggle,
  ...rest
}) => {
  const [fontRatio, setRatio] = useState(1)
  const initialFocus = useRef<boolean>(false)

  const normalizeAmount = (value, prevValue, allValues) => {
    if (Number.isNaN(Number(value)) && value !== '.' && value !== '') return prevValue
    return formatTextAmount(value, allValues && allValues.fix === 'FIAT')
  }

  const { coinfig } = window.coins[coin || 'BTC']

  const resizeSymbol = (isFiat, inputNode, fontSizeRatio, fontSizeNumber) => {
    if (Number(fontSizeRatio) > 0) {
      setRatio(fontSizeRatio > 1 ? 1 : fontSizeRatio)
    }
    const amountRowNode = inputNode.closest('#amount-row')
    const currencyNode = isFiat
      ? amountRowNode.children[0]
      : amountRowNode.children[amountRowNode.children.length - 2]
    currencyNode.style.fontSize = `${fontSizeNumber * (fontRatio - 0.3)}px`
  }

  const handleOnFocus: FocusEventHandler<HTMLInputElement> = useCallback(
    (event) => {
      if (initialFocus.current === false && event.target.value === '0') {
        event.target.value = ''
      }

      initialFocus.current = true
    },
    [initialFocus]
  )

  return (
    <AmountContainer>
      <AmountRow isError={!!amountError}>
        <div id='amount-row'>
          {fix === 'FIAT' && (
            <Text size='56px' color={amountError ? 'red400' : 'textBlack'} weight={500}>
              {Currencies[fiatCurrency].units[fiatCurrency].symbol}
            </Text>
          )}
          <Field
            autoComplete='off'
            data-e2e={rest['data-e2e']}
            name={name}
            // @ts-ignore
            component={AmountTextBox}
            normalize={normalizeAmount}
            onChange={onChange}
            onFocus={handleOnFocus}
            // eslint-disable-next-line
            onUpdate={resizeSymbol.bind(null, fix === 'FIAT')}
            maxFontSize='56px'
            placeholder='0'
            // leave fiatActive always to avoid 50% width in HOC?
            fiatActive
            haveError={!!amountError}
            {...{
              autoFocus: true,
              hideError: true
            }}
          />

          {fix === 'CRYPTO' && (
            <Text size='56px' color={amountError ? 'red400' : 'textBlack'} weight={500}>
              {coinfig.displaySymbol}
            </Text>
          )}
          {showToggle ? (
            <Icon
              color='blue600'
              cursor
              style={{ marginLeft: '12px' }}
              name='up-down-chevron'
              onClick={onToggleFix}
              role='button'
              size='24px'
              data-e2e='sendSwitchIcon'
            />
          ) : null}
        </div>
      </AmountRow>
      <QuoteRow>
        {showCounter ? (
          <>
            <Text
              color={amountError ? 'red400' : 'grey600'}
              size='14px'
              weight={500}
              data-e2e='sendQuoteAmount'
            >
              {fix === 'FIAT' && coinfig.displaySymbol} {quote} {fix === 'CRYPTO' && fiatCurrency}
            </Text>
          </>
        ) : null}
      </QuoteRow>
    </AmountContainer>
  )
}

export type Props = {
  amountError: string | boolean
  'data-e2e': string
  fiatCurrency: string
  fix: 'FIAT' | 'CRYPTO'
  name: string
  onChange?: () => void
} & (
  | // FIAT <=> COIN TOGGLE
  {
      coin: string
      fiatCurrency: string
      onToggleFix: () => void
      showToggle: true
    }
  // ONLY FIAT
  | {
      coin?: never
      fiatCurrency: string
      onToggleFix?: never
      showToggle: false
    }
) &
  (
    | {
        quote: string
        showCounter: true
      }
    | { quote?: never; showCounter: false }
  )

export default AmountFieldInput
