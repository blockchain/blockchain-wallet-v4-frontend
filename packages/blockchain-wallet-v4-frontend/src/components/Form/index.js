import AutosizeTextBox from './AutosizeTextBox'
import CaptchaBox from './CaptchaBox'
import CheckBox from './CheckBox'
import CodeInputBoxes from './CodeInputBoxes'
import CountdownTimer from './CountdownTimer'
import CreatableInputField from './CreatableInputField'
import DateBox from './DateBox'
import DateBoxDebounced from './DateBoxDebounced'
import DateInputBox from './DateInputBox'
import EmailVerification from './EmailVerification'
import FaqMessage from './FaqMessage'
import FiatConvertor from './FiatConvertor'
import FooterShadowWrapper from './FooterShadowWrapper'
import Form from './Form'
import FormError from './FormError'
import FormGroup from './FormGroup'
import FormItem from './FormItem'
import FormLabel from './FormLabel'
import NumberBox from './NumberBox'
import NumberBoxDebounced from './NumberBoxDebounced'
import PasswordBox from './PasswordBox'
import PhoneNumberBox from './PhoneNumberBox'
import RadioButton from './RadioButton'
import SelectBoxBankAccountType from './SelectBoxBankAccountType'
import SelectBox from './SelectBox'
import SelectBoxBtcAddresses from './SelectBoxBtcAddresses'
import SelectBoxBchAddresses from './SelectBoxBchAddresses'
import SelectBoxBsvAddresses from './SelectBoxBsvAddresses'
import SelectBoxEthAddresses from './SelectBoxEthAddresses'
import SelectBoxXlmAddresses from './SelectBoxXlmAddresses'
import SelectBoxCoin from './SelectBoxCoin'
import SelectBoxCoinifyCurrency from './SelectBoxCoinifyCurrency'
import SelectBoxCountry from './SelectBoxCountry'
import SelectBoxCurrency from './SelectBoxCurrency'
import SelectBoxLanguages from './SelectBoxLanguages'
import SelectBoxRegion from './SelectBoxRegion'
import SelectBoxTheme from './SelectBoxTheme'
import SelectBoxUSState from './SelectBoxUSState'
import TabMenuBuySellStatus from './TabMenuBuySellStatus'
import TabMenuTransactionStatus from './TabMenuTransactionStatus'
import TextArea from './TextArea'
import TextAreaDebounced from './TextAreaDebounced'
import TextBox from './TextBox'
import TextBoxDebounced from './TextBoxDebounced'

export {
  AutosizeTextBox,
  CaptchaBox,
  CheckBox,
  CodeInputBoxes,
  CountdownTimer,
  CreatableInputField,
  DateBox,
  DateBoxDebounced,
  DateInputBox,
  EmailVerification,
  FaqMessage,
  FiatConvertor,
  FooterShadowWrapper,
  Form,
  FormError,
  FormGroup,
  FormItem,
  FormLabel,
  NumberBox,
  NumberBoxDebounced,
  PasswordBox,
  PhoneNumberBox,
  RadioButton,
  SelectBoxBankAccountType,
  SelectBox,
  SelectBoxBtcAddresses,
  SelectBoxBchAddresses,
  SelectBoxBsvAddresses,
  SelectBoxEthAddresses,
  SelectBoxXlmAddresses,
  SelectBoxCoin,
  SelectBoxCoinifyCurrency,
  SelectBoxCountry,
  SelectBoxCurrency,
  SelectBoxLanguages,
  SelectBoxRegion,
  SelectBoxTheme,
  SelectBoxUSState,
  TabMenuBuySellStatus,
  TabMenuTransactionStatus,
  TextArea,
  TextAreaDebounced,
  TextBox,
  TextBoxDebounced
}
