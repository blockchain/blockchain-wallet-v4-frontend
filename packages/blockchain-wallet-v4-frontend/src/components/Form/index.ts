import AutosizeTextBox from './AutosizeTextBox'
import CaptchaBox from './CaptchaBox'
import CheckBox from './CheckBox'
import CodeInputBoxes from './CodeInputBoxes'
import CoinAccountListBalance from './CoinAccountListBalance'
import CoinAccountListOption from './CoinAccountListOption'
import CoinBalanceDropdown from './CoinBalanceDropdown'
import CountdownTimer from './CountdownTimer'
import CreatableInputField from './CreatableInputField'
import CreditCardBox from './CreditCardBox'
import CreditCardCVCBox from './CreditCardCVCBox'
import CreditCardExpiryBox from './CreditCardExpiryBox'
import DateBox from './DateBox'
import DateBoxDebounced from './DateBoxDebounced'
import DateInputBox from './DateInputBox'
import EmailVerification from './EmailVerification'
import FaqMessage from './FaqMessage'
import FiatConverter from './FiatConverter'
import FooterShadowWrapper from './FooterShadowWrapper'
import Form from './Form'
import FormError from './FormError'
import FormGroup from './FormGroup'
import FormItem from './FormItem'
import FormLabel from './FormLabel'
import NumberBox from './NumberBox'
import NumberBoxDebounced from './NumberBoxDebounced'
import PasswordBox from './PasswordBox'
import PhoneNumberBox from './PhoneNumberBox'
import RadioButton from './RadioButton'
import SelectBox from './SelectBox'
import SelectBoxBankAccountType from './SelectBoxBankAccountType'
import SelectBoxBchAddresses from './SelectBoxBchAddresses'
import SelectBoxBtcAddresses from './SelectBoxBtcAddresses'
import SelectBoxCoin from './SelectBoxCoin'
import SelectBoxCoinPriceChart from './SelectBoxCoinPriceChart'
import SelectBoxCountry from './SelectBoxCountry'
import SelectBoxCurrency from './SelectBoxCurrency'
import SelectBoxEthAddresses from './SelectBoxEthAddresses'
import SelectBoxLanguages from './SelectBoxLanguages'
import SelectBoxRegion from './SelectBoxRegion'
import SelectBoxTheme from './SelectBoxTheme'
import SelectBoxUSState from './SelectBoxUSState'
import SelectBoxXlmAddresses from './SelectBoxXlmAddresses'
import TabMenuTransactionStatus from './TabMenuTransactionStatus'
import TextArea from './TextArea'
import TextAreaDebounced from './TextAreaDebounced'
import TextBox from './TextBox'
import TextBoxDebounced from './TextBoxDebounced'

export {
  AutosizeTextBox,
  CaptchaBox,
  CheckBox,
  CodeInputBoxes,
  CoinAccountListBalance,
  CoinAccountListOption,
  CoinBalanceDropdown,
  CountdownTimer,
  CreatableInputField,
  CreditCardBox,
  CreditCardCVCBox,
  CreditCardExpiryBox,
  DateBox,
  DateBoxDebounced,
  DateInputBox,
  EmailVerification,
  FaqMessage,
  FiatConverter,
  FooterShadowWrapper,
  Form,
  FormError,
  FormGroup,
  FormItem,
  FormLabel,
  NumberBox,
  NumberBoxDebounced,
  PasswordBox,
  PhoneNumberBox,
  RadioButton,
  SelectBox,
  SelectBoxBankAccountType,
  SelectBoxBchAddresses,
  SelectBoxBtcAddresses,
  SelectBoxCoin,
  SelectBoxCoinPriceChart,
  SelectBoxCountry,
  SelectBoxCurrency,
  SelectBoxEthAddresses,
  SelectBoxLanguages,
  SelectBoxRegion,
  SelectBoxTheme,
  SelectBoxUSState,
  SelectBoxXlmAddresses,
  TabMenuTransactionStatus,
  TextArea,
  TextAreaDebounced,
  TextBox,
  TextBoxDebounced
}
