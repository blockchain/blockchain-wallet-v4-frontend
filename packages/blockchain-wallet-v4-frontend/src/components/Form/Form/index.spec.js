import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Form from './index.js'

describe('Form', () => {
  it('renders correctly', () => {
    const props = { override: true, onSubmit: jest.fn() }
    const component = shallow(<Form {...props} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
