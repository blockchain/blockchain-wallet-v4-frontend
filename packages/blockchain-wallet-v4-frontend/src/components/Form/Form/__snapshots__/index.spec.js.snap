// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<Form__BaseForm
  onSubmit={[MockFunction]}
/>
`;
